OpenTracing/OpenTracing.OpenTracing/ISpan.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace OpenTracing
+{
+    public interface ISpan<T> 
+    {
+        T GetSpanContext();
+
+        void Finish();
+        void FinishWithOptions(DateTime finishTime);
+
+        void SetBaggageItem(string restrictedKey, string value);
```

Comment:
unlike SetTag and Log, SetBaggageItem is meant to be paired with a GetBaggageItem...

also, I am fairly confident we are going to remove restrictions on the keys.